### PR TITLE
chore: update juju/cmd/v3 to v3.0.15

### PR DESCRIPTION
**Pull request by gfouillet**
- Before this commit, documentation generation was broken due to a bug on juju/cmd: short flag were suffixed by two dashes in documentation, instead of one (ex: `--B, -no-browser-login`)
- After this commit, documentation generation is fixed regarding that issue (ex: `-B, -no-browser-login`)

See also: https://github.com/juju/cmd/releases/tag/v3.0.15

## Checklist

- ~[ ] Code style: imports ordered, good names, simple structure, etc~
- ~[ ] Comments saying why design decisions were made~
- ~[ ] Go unit tests, with comments saying what you're testing~
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```sh
juju documentation --split --out=./docs
```
Check that the mono-character flags in generated markdown are prefixed by only one dash.

## Documentation changes

Will allows to regenerate the doc correctly.

## Links

**Jira card:** [JUJU-6296](https://warthogs.atlassian.net/browse/JUJU-6319), JUJU-6296](https://warthogs.atlassian.net/browse/JUJU-6319)

[JUJU-6296]: https://warthogs.atlassian.net/browse/JUJU-6296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by nvinuesa** [comment]
Why aren't we targetting 3.4? Was the doc gen introduced only in 3.5?

File: go.sum [hunk 1]
```diff
@@ -481,6 +481,8 @@ github.com/juju/cmd v0.0.0-20171107070456-e74f39857ca0/go.mod h1:yWJQHl73rdSX4DH
 github.com/juju/cmd/v3 v3.0.0-20220202061353-b1cc80b193b0/go.mod h1:EoGJiEG+vbMwO9l+Es0SDTlaQPjH6nLcnnc4NfZB3cY=
 github.com/juju/cmd/v3 v3.0.14 h1:KuuamArSH7vQ6SdQKEHYK2scEMkJTEZKLs8abrlW3XE=
 github.com/juju/cmd/v3 v3.0.14/go.mod h1:lGtDvm2BG+FKnIS8yY/vrhxQNX9imnL6bPIYGSTchuI=
```
Comment (nvinuesa) [inline go.sum:483]: These two lines will be removed if you run `go mod tidy`, this is what's making the static and generate checks fail.

Comment (gfouillet) [reply go.sum]: Yes this is linted anyway. fixed it.

**Review by nvinuesa — APPROVED** [review]
LGTM, thanks! QA passes well.

**Pull request merged**